src/shared/infra/http/controllers/registry-offices.controller.ts

```diff
@@ -0,0 +1,116 @@
+import { CreateRegistryOffice } from '@modules/registry-office/use-cases/create-registry-office';
+import { GetRegistryOfficeById } from '@modules/registry-office/use-cases/get-registry-office-by-id';
+import { GetRegistryOfficeByName } from '@modules/registry-office/use-cases/get-registry-office-by-name';
+import { Body, Controller, Get, Param, Post, Put } from '@nestjs/common';
+import { Public } from '@shared/utils/public-decorator';
+import { RegistryOfficeViewModel } from '../view-models/registry-office-view-model';
+import { CreateRegistryOfficeBody } from '../dtos/create-registry-office-body';
+import { DeleteRegistryOffice } from '@modules/registry-office/use-cases/delete-registry-office';
+import { UpdateRegistryOffice } from '@modules/registry-office/use-cases/update-registry-office';
+import { UpdateRegistryOfficeBody } from '../dtos/update-registry-office-body';
+
+
+
+@Controller('registry-offices')
+export class RegistryOfficesController {
+  constructor(
+    private createRegistryOffice: CreateRegistryOffice,
+    private getRegistryOfficeById: GetRegistryOfficeById,
+    private getRegistryOfficeByName: GetRegistryOfficeByName,
+    private deleteRegistryOffice: DeleteRegistryOffice,
+    private updateRegistryOffice: UpdateRegistryOffice,
+  ) {}
+
+  @Post()
+  @Public()
+  async create(@Body() body: CreateRegistryOfficeBody) {
+    const {
+      name,
+      description,
+      logo,
+      address_id,
+      document,
+      phone,
+      created_at,
+      updated_at,
+    } = body;
+
+    const { registry_office } = await this.createRegistryOffice.execute({
+      name,
+      logo,
+      description,
+      address_id,
+      document,
+      phone,
+      created_at,
+      updated_at,
+    });
+
+    return {
+      registry_office: RegistryOfficeViewModel.toHTTP(registry_office),
+    };
+  }
+
+  @Get(':/id')
+  @Public()
+  async GetRegistryOfficeId(@Param('id') id) {
```

Comment:
sempre que colocar o parametro id ou qualquer outro o node por si só não tem tipagem, como estamos utilizando typescript temos que forçar o tipo para o mesmo.
async GetRegistryOfficeId(@Param('id') **id**) 

Ficando algo assim:
async GetRegistryOfficeId(@Param('id') id: string)